theano/gpuarray/opt.py

```diff
@@ -235,10 +250,10 @@ def local_opt(node):
                     elif isinstance(new_op, (tuple, list)):
                         return [safe_to_cpu(o) for o in new_op]
                     else:  # suppose it is a variable on the GPU
-                        return [host_from_gpu(new_op)]
+                        return [new_op.transfer('cpu')]
             return False
         local_opt.__name__ = maker.__name__
-        return local_optimizer(OP)(local_opt)
+        return register_topo('fast_compile')(local_optimizer(OP)(local_opt))
```

Comment:
For now, I think we should not change the logic of move_to_gpu. Maybe it could be a Theano flags, but I would not change the default for now. There is cases with subtensor where we want to keep them on the CPU.

Can you point me to the example that would show a speed up from a fast back-ward pass? If I see a profile that show it is useful, we can spin that to another PR.